### ISSUE TITLE
bug: webpack import error after upgrade to v4.0.3 (from 3.9.4)

### ISSUE THREAD

**Issue by Goldziher** [saadeghi/daisyui#2517]
### What version of daisyUI are you using?

4.0.3

### Which browsers are you seeing the problem on?

All browsers

### Reproduction URL

-

### Describe your issue

Hi, 

So i updated the library to the latest version, and its now failing to build. I get this:

```shell
🌼   daisyUI 4.0.3
  ⨯ ./src/styles/globals.scss.webpack[javascript/auto]!=!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[12].oneOf[13].use[2]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[12].oneOf[13].use[3]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/resolve-url-loader/index.js??ruleSet[1].rules[12].oneOf[13].use[4]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/compiled/sass-loader/cjs.js??ruleSet[1].rules[12].oneOf[13].use[5]!./src/styles/globals.scss
 TypeError: Cannot read properties of undefined (reading 'l')
     at Array.map (<anonymous>)
     at Array.reduce (<anonymous>)
     at Array.forEach (<anonymous>)
     at Array.forEach (<anonymous>)
     at Array.forEach (<anonymous>)
 Import trace for requested module:
 ./src/styles/globals.scss.webpack[javascript/auto]!=!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[12].oneOf[13].use[2]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[12].oneOf[13].use[3]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/build/webpack/loaders/resolve-url-loader/index.js??ruleSet[1].rules[12].oneOf[13].use[4]!../node_modules/.pnpm/next@14.0.2_@babel+core@7.23.3_react-dom@18.2.0_react@18.2.0_sass@1.69.5/node_modules/next/dist/compiled/sass-loader/cjs.js??ruleSet[1].rules[12].oneOf[13].use[5]!./src/styles/globals.scss
 ./src/styles/globals.scss
```

**Comment by petermazzocco** [comment]
Getting the same error, just started with 4.0.3

**Comment by saadeghi** [comment]
Can you update to version `4.0.4` and try again?

And please provide a reproduction repo, otherwise it's hard to guess what is causing the issue.

**Comment by Goldziher** [comment]
> Can you update to version `4.0.4` and try again?

Nope - still the same issue.

> And please provide a reproduction repo, otherwise it's hard to guess what is causing the issue.

The issue is simply nextjs 14 + daisyui 4. I would recommend adding an e2e test in github actions that ensures next versions can build.

**Comment by federico-smai** [comment]
> > Can you update to version `4.0.4` and try again?
> 
> Nope - still the same issue.
> 
> > And please provide a reproduction repo, otherwise it's hard to guess what is causing the issue.
> 
> The issue is simply nextjs 14 + daisyui 4. I would recommend adding an e2e test in github actions that ensures next versions can build.

Same

**Comment by olsommer** [comment]
Same here

**Comment by vinayman** [comment]
Same here

**Comment by LiamKenneth** [comment]
haha same issue but I was using version 4.0.5 and Next.js 13.5 so it's not specific to next.js 14

**Comment by saadeghi** [comment]
6 people with same issue and no one shares a reproduction repo…

Here's an example of a fresh install Next.js 14 + daisyUI 4:  
https://github.com/saadeghi/nextjs14-daisyui4 (look at [tailwind config](https://github.com/saadeghi/nextjs14-daisyui4/blob/main/tailwind.config.cjs))


We can't update [the StackBlitz example](https://stackblitz.com/edit/daisyui-nextjs) to Next.js 14 yet because Next.js 14 doesn't work in StackBlitz.

Let me know if you have any questions.

**Comment by Truimo** [comment]
I have this problem here, but I haven't solved it. If possible, please help to check it out, thanks.

https://github.com/Truimo/home/tree/renovate/daisyui-4.x

**Comment by saadeghi** [comment]
>TL;DR instead of [data-theme=light] just pass in the string value of the theme "light"

Yes. I added this to the v4 breaking changes yesterday (it was missing) and the docs are updated.

**Comment by santi020k** [comment]
My config:

```tsx
import daisyuiTheme from 'daisyui/src/theming/themes'

...
light: {
      ...daisyuiTheme?.light,
      primary: lightColors.PRIMARY,
      }
...
```

@saadeghi By the way, very good update, I love working with this library with astro.
